parser.temper

```diff
@@ -9,8 +11,8 @@ class Parser {
     str: String,
     slots: Mapped<String, RegexNode> = new Map([]),
   ) {
-    chars = str.split("");
-    pos = 0;
+    chars = str;
+    pos = String.start;
```

Comment:
Could probably rename the field from `chars` to `str`, but I did minimal changes in some cases here.